app/parse.py

```diff
@@ -1,15 +1,64 @@
+import csv
+import requests
+from bs4 import BeautifulSoup
 from dataclasses import dataclass
+from typing import List
+
+
+BASE_URL = "https://quotes.toscrape.com/"
 
 
 @dataclass
 class Quote:
     text: str
     author: str
-    tags: list[str]
+    tags: List[str]
+
+
+def parse_quote(quote_soup) -> Quote:
+    text = quote_soup.select_one(".text").get_text(strip=True)
+    author = quote_soup.select_one(".author").get_text(strip=True)
+    tags = [tag.get_text(strip=True) for tag in quote_soup.select(".tag")]
+    return Quote(text, author, tags)
+
+
+def get_all_quotes() -> List[Quote]:
+    quotes = []
+    page_num = 1
+
+    while True:
+        response = requests.get(f"{BASE_URL}/page/{page_num}")
```

Comment:
Consider adding error handling for the HTTP request using `requests.get`. This could involve checking the status code of the response and handling exceptions such as `requests.exceptions.RequestException` to ensure the program doesn't crash if the request fails.